addons/settings/fnc_initDisplayMain.sqf

```diff
@@ -16,18 +16,7 @@ if (isClass (configFile >> "CfgPatches" >> QGVAR(userconfig))) then {
 private _userconfig = "";
 
 if (_file != "") then {
-    private _fileExists = false;
-
-    if (!isNull _display) then {
-        private _control = _display ctrlCreate ["RscHTML", -1];
-        _control htmlLoad _file;
-        _fileExists = ctrlHTMLLoaded _control;
-        ctrlDelete _control;
-    } else {
-        _fileExists = loadFile _file != "";
-    };
-
-    if (_fileExists) then {
+    if (FILE_EXISTS(_file)) then {
```

Comment:
`initDisplayMain.sqf` is executed in the onLoad event of RscDisplayMain/idd=0. `findDisplay 0` will report null, because `findDisplay` is bugged and only reports displays that were created at least one frame* earlier.
This will cause the macro script to fall back to `loadFile`. And if `loadFile` fails, it will create an error message.
This leads to this branch of CBA to create an error pop up when loaded without cba_settings.sqf file in the main directory and file patching enabled. Try it.

Reply:
Should this stay how it was or should there be a macro like `FILE_EXISTS_CUSTOM` also accepting a display as param to handle such cases?

Reply:
Could read the main display from uiNamespace:
```sqf
uiNamespace getVariable "RscDisplayMain"
```
That should work even during this onLoad event as XEH DisplayLoaded is executed after the BIS_fnc_initDisplay script that sets these variables.
You should also drop the 313 display completely, as the main menu display does exist in 3den.

Reply:
Does the main menu also exist when a mission is loaded? If not, does display 313 exist there? How would this behave when neither in 3DEN nor in main menu?

Reply:
I'm pretty sure the main menu display survives the whole session, but not 100% as I haven't messed around with it a lot in (dedicated client) MP.
Display3den is created once you enter the editor and it survives even through previews until you leave the editor back to the main menu.

Reply:
After testing around a bit it seems like it really preserves the whole session and even in (dedicated) MP. Should work in all cases.

Reply:
Good point. But just calling `disableSerialization` would have the side effect of disabling serilization for the script. Which cannot be intended.
Should wrap the thing into an `isNil` block to force unscheduled environment I'd say. Or alternatively store the control inside an array or something.

Reply:
Isn't `disableSerialization` scope based? However, in my tests I got no error without `disableSerialization` which is strange but the macro still seems to work fine. And I think you also didn't use `disableSerialization` in the settings scoped macro.

Reply:
@chris579 Script based. VM based.

Reply:
>Isn't disableSerialization scope based?

No.

>in my tests I got no error

The error requires serialization to be enabled in the first place. Local variables in unscheduled environment are not affected.

Reply:
>And I think you also didn't use disableSerialization in the settings scoped macro.

Yes, because it was for this specific component, and I don't ever use the scheduled environment.